test-network-function/networking/suite.go

```diff
@@ -188,23 +194,52 @@ func processContainerIpsPerNet(containerID *configsections.ContainerIdentifier,
 	netsUnderTest[netKey] = entry
 }
 
+func IsIPv4(address string) bool {
+	return strings.Count(address, ":") < 2 //nolint:gomnd // less readable
+}
+
+func IsIPv6(address string) bool {
+	return strings.Count(address, ":") >= 2 //nolint:gomnd // less readable
+}
+
+func FilterIPListPerVersion(ipList []string, ipVersion string) []string {
+	var filteredIPList []string
+	for _, aIP := range ipList {
+		if ipVersion == IPv4 && IsIPv4(aIP) {
+			filteredIPList = append(filteredIPList, aIP)
+		}
+		if ipVersion == IPv6 && IsIPv6(aIP) {
+			filteredIPList = append(filteredIPList, aIP)
+		}
+	}
+	return filteredIPList
+}
+
 // runNetworkingTests takes a map netTestContext, e.g. one context per network attachment
 // and runs pings test with it. Returns a network name to a slice of bad target IPs map.
-func runNetworkingTests(netsUnderTest map[string]netTestContext, count int) map[string][]string {
+func runNetworkingTests(netsUnderTest map[string]netTestContext, count int, ipVersion string) map[string][]string {
```

Comment:
ipVersion should be better to be defined as new type and have two values ? or enum

Reply:
fixed